services/blockchain-indexer/app.js

```diff
@@ -49,33 +53,65 @@ const app = Microservice({
 	dependencies: [
 		'connector',
 	],
-});
+};
 
-setAppContext(app);
+const app = Microservice(defaultBrokerConfig);
 
-// Add routes, events & jobs
-if (config.operations.isIndexingModeEnabled) {
-	app.addMethods(path.join(__dirname, 'methods', 'indexer'));
-	app.addEvents(path.join(__dirname, 'events'));
-}
+// Start a temporary broker to query for SDK module names
+// To be used for dynamically registering the available module specific endpoints
+const tempApp = Microservice({
+	...defaultBrokerConfig,
+	name: 'temp_service_indexer',
+	events: {},
+});
+setAppContext(tempApp);
 
-if (config.operations.isDataRetrievalModeEnabled) {
-	app.addMethods(path.join(__dirname, 'methods', 'dataService'));
-	app.addJobs(path.join(__dirname, 'jobs', 'dataService'));
-}
+tempApp.run().then(async () => {
+	const { getRegisteredModules } = require('./shared/constants');
+	const registeredModules = await getRegisteredModules();
+	registeredModules.forEach((module, index) => {
+		// Map 'reward' module to the 'dynamicReward' module endpoints
+		if (module === MODULE.REWARD) registeredModules[index] = module.DYNAMIC_REWARD;
+	});
```

Comment:
Please consider moving this logic below just above: `registeredModules.forEach(module => {` on L89.

Reply:
Applied.